light/errors.go

```diff
@@ -20,6 +53,134 @@ func (e ErrOldHeaderExpired) Error() string {
+}
+
+type ErrHeaderHeightNotMonotonic struct {
+	WantHeight int64
+	GetHeight  int64
+}
+
+func (e ErrHeaderHeightNotMonotonic) Error() string {
+	return fmt.Sprintf("expected new header height %d to be greater than one of old header %d", e.WantHeight, e.GetHeight)
+}
+
+type ErrHeaderTimeNotMonotonic struct {
+	WantTime time.Time
+	GetTime  time.Time
+}
+
+func (e ErrHeaderTimeNotMonotonic) Error() string {
+	return fmt.Sprintf("expected new header time %v to be after old header time %v", e.WantTime, e.GetTime)
+}
+
+type ErrHeaderTimeExceedMaxClockDrift struct {
+	Ti    time.Time
+	Now   time.Time
+	Drift time.Duration
+}
+
+func (e ErrHeaderTimeExceedMaxClockDrift) Error() string {
+	return fmt.Sprintf("new header has a time from the future %v (now: %v; max clock drift: %v)", e.Ti, e.Now, e.Drift)
+}
+
+type ErrUnverifiedHeight struct {
+	Height int64
+}
+
+func (e ErrUnverifiedHeight) Error() string {
+	return fmt.Sprintf("unverified header/valset requested (latest: %d)", e.Height)
+}
+
+type ErrInvalidTrustLevel struct {
+	Level cmtmath.Fraction
+}
+
+func (e ErrInvalidTrustLevel) Error() string {
+	return fmt.Sprintf("trustLevel must be within [1/3, 1], given %v", e.Level)
+}
+
+type ErrValidatorsMismatch struct {
+	HeaderHash     cmtbytes.HexBytes
+	ValidatorsHash []byte
+	Height         int64
+}
+
+func (e ErrValidatorsMismatch) Error() string {
+	return fmt.Sprintf("expected new header validators (%X) to match those that were supplied (%X) at height %d", e.HeaderHash, e.ValidatorsHash, e.Height)
+}
+
+type ErrValidatorHashMismatch struct {
+	TrustedHash   cmtbytes.HexBytes
+	ValidatorHash cmtbytes.HexBytes
+}
+
+func (e ErrValidatorHashMismatch) Error() string {
+	return fmt.Sprintf("expected old header next validators (%X) to match those from new header (%X)", e.TrustedHash, e.ValidatorHash)
+}
+
+type ErrBlockHashMismatch struct {
+	TraceBlockHash  cmtbytes.HexBytes
+	SourceBlockHash cmtbytes.HexBytes
+}
+
+func (e ErrBlockHashMismatch) Error() string {
+	return fmt.Sprintf("trusted block is different to the source's first block (%X = %X)", e.TraceBlockHash, e.SourceBlockHash)
+}
+
+type ErrHeaderHashMismatch struct {
+	Want cmtbytes.HexBytes
+	Get  cmtbytes.HexBytes
+}
+
+func (e ErrHeaderHashMismatch) Error() string {
+	return fmt.Sprintf("expected header's hash %X, but got %X", e.Want, e.Get)
+}
+
+type ErrExistingHeaderHashMismatch struct {
+	Existing cmtbytes.HexBytes
+	New      cmtbytes.HexBytes
+}
+
+func (e ErrExistingHeaderHashMismatch) Error() string {
+	return fmt.Sprintf("existing trusted header %X does not match newHeader %X", e.Existing, e.New)
+}
+
+type ErrLightHeaderHashMismatch struct {
+	Light cmtbytes.HexBytes
+	New   cmtbytes.HexBytes
+}
+
+func (e ErrLightHeaderHashMismatch) Error() string {
+	return fmt.Sprintf("light block header %X does not match newHeader %X", e.Light, e.New)
+}
+
+type ErrInvalidHashSize struct {
+	Want int
+	Got  int
```

Comment:
```suggestion
	Expected int
	Actual int
```

I trust that we need to soft-code the expected hash size because we might (some day) have differently sized hashes.
Otherwise, it's a woeful waste of an extra integer field to this Rust programmer :grin: